### DISCUSSION TITLE
open sourcing the ts types generation code

### DISCUSSION THREAD

**Discussion by a7madgamal** [js-fns/date-fns#1799] [General] [answered]
I'm trying to build a components library and have similar ts structure and types like date-fns as an example but I found the sophisticated ts build code. is it something that can be extracted into a separate npm package?

**Comment by kossnocorp** [comment, marked as answer]
It is something that can be extracted, but we won't work on that. We're migrating to TypeScript and not interested in maintaining such a library.

Given that you're welcome to extract and publish it yourself.

**Reply by a7madgamal** [reply]
wait, I thought it was using TS under the hood 🤦‍♂️
thanks the the quick answer 👍🏻👍🏻

**Reply by kossnocorp** [reply]
No, we actually generate TS as well as Flow typings from JSDoc annotations.
